conjure-java-core/src/main/java/com/palantir/conjure/java/types/UnionGenerator.java

```diff
@@ -173,6 +174,40 @@ private static MethodSpec generateGetValue(ClassName baseClass) {
                     return builder.build();
                 })
                 .collect(Collectors.toList());
+        staticFactories.add(generateUnknownStaticFactory(unionClass, memberTypeDefs));
+        return staticFactories;
+    }
+
+    private static MethodSpec generateUnknownStaticFactory(ClassName unionClass, List<FieldDefinition> memberTypeDefs) {
+        String typeParam = "type";
+        String valueParam = "value";
+        MethodSpec.Builder builder = MethodSpec.methodBuilder("unknown")
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                .addParameter(String.class, typeParam)
+                .addParameter(Object.class, valueParam)
+                .returns(unionClass);
+        // begin switch statement
+        builder.beginControlFlow("switch($L)", Expressions.requireNonNull(typeParam, "Type is required"));
+        // add all cases
+        memberTypeDefs.forEach(memberTypeDef -> {
+            String memberName = sanitizeUnknown(memberTypeDef.getFieldName()).get();
+            builder.addCode("case $S:", memberName);
+            builder.addStatement(
+                    "throw new $T(\"$L: $L\")",
+                    SafeIllegalArgumentException.class,
+                    "Unknown type cannot be created as the provided type is known",
+                    memberName);
```

Comment:
Sorry, I should have caught this before:

Let's let javapoet handle strings for us
```suggestion
                    "throw new $T($S)",
                    SafeIllegalArgumentException.class,
                    "Unknown type cannot be created as the provided type is known: " + memberName);
```